### PR TITLE
Create a Jenkins Job that creates an inventory list

### PR DESCRIPTION
**Pull request by samuel-rubin**
This is a Jenkins job that gets all the nodes that fit a specified label and do multiple things:
-	It will create an inventory ini file that is readable by ansible tower
-	It will create a markdown table that gives a summary of the machines
-	It will send a slack message listing all of the machines that are offline and their reason

This job will also push the inventory ini and summary table to either push them to git
 or it will archive them.

Signed-off-by: Samuel Rubin <samuel.rubin@ibm.com>

### CODE REVIEW

**Comment by samuel-rubin** [comment]
This is could be related to #25

**Comment by samuel-rubin** [comment]
@AdamBrousseau

**Comment by karianna** [comment]
Paging @sxa555 - This ties into infra

File: Jenkins_jobs/GetInventoryList.groovy [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+CREATE_INVENTORY_INI: Boolean - This toggles if the job will create the inventory ini file
+CREATE_INVENTORY_SUMMARY: Boolean - This toggles if the summary talbe will be generated
+SLACK_CHANNEL: String - What channel a list of problematic machies gets sent to. If this is empty, no slack message will be sent
+SSH_CREDENTIALS: Credentials - The credentials needed to push the files to git
+GIT_REPO: String - The git repo to push the files to. If this is not set, the files will be archived to Jenkins
+GIT_BRANCH: String - The git branch to push the files to. It defaults to master
+GIT_FOLDER: String - The folder that the files will be saved to. It defaults to a null string
+*/
+
+@Library('NodeHelper-Sam') _
```
Comment (jdekonin) [inline Jenkins_jobs/GetInventoryList.groovy]: Might want to remove your name here. :)

File: Jenkins_jobs/GetInventoryList.groovy [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Parameters: 
+SETUP_LABEL: String - the label of the node that this job runs on. The default is worker
+LABEL: String - the label of the machines that this job will look for
+CREATE_INVENTORY_INI: Boolean - This toggles if the job will create the inventory ini file
+CREATE_INVENTORY_SUMMARY: Boolean - This toggles if the summary talbe will be generated
```
Comment (jdekonin) [inline Jenkins_jobs/GetInventoryList.groovy]: spelling of `table`

File: Jenkins_jobs/GetInventoryList.groovy [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+SLACK_CHANNEL: String - What channel a list of problematic machies gets sent to. If this is empty, no slack message will be sent
+SSH_CREDENTIALS: Credentials - The credentials needed to push the files to git
```
Comment (jdekonin) [inline Jenkins_jobs/GetInventoryList.groovy]: spelling or `machines`

File: Jenkins_jobs/GetInventoryList.groovy [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Parameters: 
+SETUP_LABEL: String - the label of the node that this job runs on. The default is worker
+LABEL: String - the label of the machines that this job will look for
```
Comment (jdekonin) [inline Jenkins_jobs/GetInventoryList.groovy]: LABEL can be used as a filter to only get specific machines; like `ci.role.performance`.    Then the output file (groovy code) will only use this subset to operate with.  Is this correct?

Comment (samuel-rubin) [reply Jenkins_jobs/GetInventoryList.groovy]: Correct

**Comment by samuel-rubin** [comment]
I fixed the spelling errors

**Pull request merged**